src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/FormPass.php

```diff
@@ -34,18 +34,19 @@ public function process(ContainerBuilder $container)
         $types = array();
 
         foreach ($container->findTaggedServiceIds('form.type') as $serviceId => $tag) {
+            // Support type access by FQCN
+            $serviceDefinition = $container->findDefinition($serviceId);
+            $types[$serviceDefinition->getClass()] = $serviceId;
+
             // The following if-else block is deprecated and will be removed
             // in Symfony 3.0
             // Deprecation errors are triggered in the form registry
             if (isset($tag[0]['alias'])) {
+                @trigger_error(sprintf('The alias option of the form.type tag is deprecated since version 2.8 and will be removed in 3.0. Use the fully-qualified type class name "%s" instead.', $serviceDefinition->getClass()));
```

Comment:
no need to trigger a deprecation warning here. It only makes it harder to support both the old and the new APIs, but having to set the tag selectively. Having an unused attribute does not hurt. And the form component will already trigger a deprecation warning in case you access the form type by its alias.